File: 240504/빙빙 돌며 사각형 채우기/snail-alphabet-square.java
import java.util.Scanner;

public class Main {
    static int[] dx = {0,1,0,-1};
    static int[] dy = {1,0,-1,0};

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        int m = sc.nextInt();
        char[][] arr = new char[n][m];
        for(int i=0;i<n;i++) {
            for(int j=0;j<m;j++) {
                arr[i][j] = 0;
            }
        }
        char ch = 'A';

        int count = n*m;
        int row = 0;
        int col = 0;
        int dir = 0;
        for(int i=0;i<count;i++) {
            arr[row][col] = ch;
            ch++;
            if(ch > 'Z') {
                ch = 'A';
            }
            int nx = row + dx[dir];
            int ny = col + dy[dir];
            if(nx < 0 || nx >= n || ny < 0 || ny >= m || arr[nx][ny] != 0) {
                dir = (dir + 1) % 4;
            }
            row += dx[dir];
            col += dy[dir];
        }
        for(int i=0;i<n;i++) {
            for(int j=0;j<m;j++) {
                System.out.print(arr[i][j] + " ");
            }
            System.out.println();
        }
    }
}

File: 240425/빙빙 돌며 사각형 채우기/snail-alphabet-square.kt
import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)

    val n = sc.nextInt()
    val m = sc.nextInt()

    val arr2D = Array(n) { CharArray(m) }

    var dirNum = 0

    val dx = intArrayOf(0, 1, 0, -1)
    val dy = intArrayOf(1, 0, -1, 0)

    var x = 0
    var y = 0

    arr2D[x][y] = 'A'

    var curAlphabet = 'A'

    for (i in 1 until n * m) {
        val nx = x + dx[dirNum]
        val ny = y + dy[dirNum]

        if (!inRange(n, m, nx, ny) || arr2D[nx][ny] in 'A'..'Z') {
            dirNum = (dirNum + 1) % 4
        }

        x += dx[dirNum]
        y += dy[dirNum]

        curAlphabet = (65 + (i % 26)).toChar()
        arr2D[x][y] = curAlphabet
    }

    for (i in 0 until n) {
        for (j in 0 until m) {
            print("${arr2D[i][j]} ")
        }
        println()
    }
}

fun inRange(n: Int, m: Int, x: Int, y: Int): Boolean {
    return (x in 0 until n) && (y in 0 until m)
}